xtable-core/src/main/java/org/apache/xtable/kernel/DeltaKernelConversionSource.java

```diff
@@ -0,0 +1,245 @@
+        tableExtractor.table(table, snapshot, engine, tableName, basePath);
+    Map<String, InternalDataFile> addedFiles = new HashMap<>();
+    String provider = ((SnapshotImpl) snapshot).getMetadata().getFormat().getProvider();
+    FileFormat fileFormat = actionsConverter.convertToFileFormat(provider);
+    List<FileStatus> files =
+        DeltaLogActionUtils.listDeltaLogFilesAsIter(
+                engine,
+                Collections.singleton(FileNames.DeltaLogFileType.COMMIT),
+                new Path(basePath),
+                versionNumber,
+                Optional.of(versionNumber),
+                false)
+            .toInMemoryList();
+
+    List<Row> actions = new ArrayList<>();
+    ActionsIterator actionsIterator =
+        new ActionsIterator(engine, files, actionSchema, Optional.empty());
+    while (actionsIterator.hasNext()) {
+      // Each ActionWrapper may wrap a batch of rows (actions)
+      CloseableIterator<Row> scanFileRows = actionsIterator.next().getColumnarBatch().getRows();
+      while (scanFileRows.hasNext()) {
+        Row scanFileRow = scanFileRows.next();
+        if (scanFileRow instanceof AddFile) {
+          Map<String, String> partitionValues =
+              InternalScanFileUtils.getPartitionValues(scanFileRow);
+          //    List<Action> actionsForVersion =
+          // getChangesState().getActionsForVersion(versionNumber);
+          InternalDataFile dataFile =
+              actionsConverter.convertAddActionToInternalDataFile(
+                  (AddFile) scanFileRow,
+                  table,
+                  fileFormat,
+                  tableAtVersion.getPartitioningFields(),
+                  tableAtVersion.getReadSchema().getFields(),
+                  true,
+                  DeltaKernelPartitionExtractor.getInstance(),
+                  DeltaKernelStatsExtractor.getInstance(),
+                  partitionValues);
+          addedFiles.put(dataFile.getPhysicalPath(), dataFile);
+        }
+      }
+    }
+
+    InternalFilesDiff internalFilesDiff =
+        InternalFilesDiff.builder().filesAdded(addedFiles.values()).build();
+    return TableChange.builder()
+        .tableAsOfChange(tableAtVersion)
+        .filesDiff(internalFilesDiff)
+        .sourceIdentifier(getCommitIdentifier(versionNumber))
+        .build();
+  }
+
+  @Override
+  public CommitsBacklog<Long> getCommitsBacklog(
+      InstantsForIncrementalSync instantsForIncrementalSync) {
+    Configuration hadoopConf = new Configuration();
+    Engine engine = DefaultEngine.create(hadoopConf);
+    Table table = Table.forPath(engine, basePath);
+    Snapshot snapshot =
+        table.getSnapshotAsOfTimestamp(
+            engine, Timestamp.from(instantsForIncrementalSync.getLastSyncInstant()).getTime());
+
+    long versionNumberAtLastSyncInstant = snapshot.getVersion();
+    System.out.println("versionNumberAtLastSyncInstant: " + versionNumberAtLastSyncInstant);
+    //    resetState(0, engine,table);
```

Comment:
This is the place which I have commented @vinishjail97 , call is happening inside this method